internal/exec/util/file.go

```diff
@@ -173,14 +182,19 @@ func (u Util) PerformFetch(f *FetchOp) error {
 		return err
 	}
 
+	// ioutil.TempFile defaults to 0600
+	if err := tmp.Chmod(DefaultFilePermissions); err != nil {
```

Comment:
Should this not be after the `defer tmp.Close()`, if the `Chmod` fails we could exit without closing.